docs/root/configuration/observability/access_log/usage.rst

```diff
@@ -1051,6 +1051,20 @@ The following command operators are supported:
 %FILTER_CHAIN_NAME%
   The :ref:`network filter chain name <envoy_v3_api_field_config.listener.v3.FilterChain.name>` of the downstream connection.
 
+%ACCESS_LOG_TYPE%
+  The type of the access log, which indicates when the access log was recorded. If a non-supported log (from the list below),
+  uses this substitution string, then the value will be an empty string.
+
+  * TcpUpstreamConnected - When TCP Proxy filter has successfully established an upstream connection.
+  * TcpPeriodic - On any TCP Proxy filter periodic log record.
+  * TcpEnd - When a TCP connection is ended on TCP Proxy filter.
+  * HcmNewRequest - When HTTP Connection Manager filter receives a new HTTP request.
+  * HcmPeriodic - On any HTTP Connection Manager periodic log record.
+  * HcmEnd - When an HTTP stream is ended on HTTP Connection Manager filter.
```

Comment:
HttpNewRequest/HttpPeriodic/HttpEnd.

Reply:
'Http' works both for HCM and router filters. If this changes to HttpNewRequest, what would be the name for the router values?

Reply:
HCM is name of HTTP L4 proxy, it's thing about the inner impl. I think it's would be better not to expose it to end users.

**And re-think about it now, IMO,  `DownstreamStart/DownstreamPeriodic/DownstreamEnd` may be better. Then this enum could be used for any L7 logging.**

> 'Http' works both for HCM and router filters. If this changes to HttpNewRequest, what would be the name for the router values?

I think both the `RouterStart/RouterPeriodic/RouterEnd` and `UpstreamStart/UpstreamPeriodic/UpstreamEnd` are ok.

Reply:
So you suggeat to rename 'Tcp' to 'Upstream', 'HCM' to 'Downstream' and keep router as-is?

Reply:
Sorry, maybe may expressions are not clear. My suggesions are:
1. `TcpUpstreamConnected/TcpPeriodic/TcpEnd` is ok for the tcp logging.
2. `DownstreamStart/DownstreamPeriodic/DownstreamEnd` for the L7 downstream logging.
3. `UpstreamStart/UpstreamPeriodic/UpstreamEnd` for the L7 upstream logging.

Reply:
``UpstreamStart`` can be logically referenced to both Router and TCP proxy, can't it? I mean, I guess that if we make this clear in the documentation that Tcp is for TCP proxy and Upstream is for router, then it should be generally ok, but if someone doesn't read or familiar with the documents, they may get confused. On the other hand, the filter chain can only have either TCP Proxy or router, so I can't see a way for a real confusion. Just sharing my thoughts here, please give a final stamp that we should rename this and I'll update.

Reply:
> Upstream is for router

I will not say Upstream is for router. `Upstream` is for the upstream request. Although for the connection managers (Http, Dubbo, Thrift), they typically use a router filter to send the upstream request.

> UpstreamStart can be logically referenced to both Router and TCP proxy, can't it?

Yeah, but I think this is a high level feature for high level users, so if we document it clear should be ok.

Reply:
done